server_utils/origin.go

```diff
@@ -758,11 +784,29 @@ func CheckOriginSentinelLocations(exports []OriginExport) (ok bool, err error) {
 			if path.Base(sentinelPath) != sentinelPath {
 				return false, errors.Errorf("invalid SentinelLocation path for StoragePrefix %s, file must not contain a directory. Got %s", export.StoragePrefix, export.SentinelLocation)
 			}
-			fullPath := filepath.Join(export.StoragePrefix, sentinelPath)
-			_, err := os.Stat(fullPath)
+
+			fullPath := filepath.Join(export.FederationPrefix, sentinelPath)
+			tkn, err := generateFileTestScitoken(sentinelPath)
+			if err != nil {
+				return false, errors.Wrap(err, "Failed to generate self-auth token for sentinel file check")
+			}
+
+			sentinelUrl := fmt.Sprintf("%v%v", param.Origin_Url.GetString(), fullPath)
```

Comment:
I'd be a bit more comfortable if this URL was constructed using the `net/url` package, unless we're certain we won't encounter funkiness with missing  or duplicate `/`s between the joined strings.